extmod/libmetal/libmetal.mk

```diff
@@ -10,7 +10,7 @@ $(BUILD)/openamp: $(BUILD)
 $(BUILD)/openamp/metal: $(BUILD)/openamp
 	$(MKDIR) -p $@
 
-$(BUILD)/openamp/metal/config.h: $(BUILD)/openamp/metal $(TOP)/$(LIBMETAL_DIR)/lib/config.h
+$(BUILD)/openamp/metal/config.h: | $(BUILD)/openamp/metal $(TOP)/$(LIBMETAL_DIR)/lib/config.h
```

Comment:
Wouldn't you want to rebuild this target if `lib/libmetal` changes, ie keep the second part a normal prerequisite?  Eg if you check out a different commit from `lib/libmetal` you'd want this rule to run again.

But probably the two mkdir rules above can be changed to order-only prerequisites.

Reply:
> ie keep the second part a normal prerequisite

I thought this is what this change is doing, i.e., make the first order-only prerequisite and the second normal prerequisite, no ?

Reply:
I can change it to the following:
```
$(BUILD)/openamp/metal/config.h: $(TOP)/$(LIBMETAL_DIR)/lib/config.h | $(BUILD)/openamp/metal
```

Reply:
Fixed.

Reply:
Yes, that looks better now.

Do you also want to make the rules above this line for `$(BUILD)/openamp` and `$(BUILD)/openamp/metal` also order-only?

Reply:
Done. I've tested the build and it's working fine. Also changing `$(TOP)/$(LIBMETAL_DIR)/lib/config.h` triggers a rebuild.